uaclient/tests/test_contract.py

```diff
@@ -96,6 +97,32 @@ def test_overrides_applied_before_comparison(
 
 @mock.patch(M_PATH + "UAContractClient")
 class TestGetAvailableResources:
+    def test_request_resources_error_on_network_disconnected(self, client):
+        """Raise error get_available_resources can't contact backend"""
+        cfg = FakeConfig()
+
+        platform = util.get_platform_info()
+        resource_params = {
+            "architecture": platform["arch"],
+            "series": platform["series"],
+            "kernel": platform["kernel"],
+        }
+        url = API_V1_RESOURCES + "?" + urllib.parse.urlencode(resource_params)
+
+        def fake_contract_client(cfg):
+            fake_client = FakeContractClient(cfg)
+            fake_client._responses = {
+                url: util.UrlError(
+                    socket.gaierror(-2, "Name or service not known")
+                )
+            }
+            return fake_client
+
+        client.side_effect = fake_contract_client
+        with pytest.raises(util.UrlError) as exc:
+            get_available_resources(cfg)
+        assert "[Errno -2] Name or service not known" == str(exc.value)
```

Comment:
Would it make sense to instead assert that `exc` is exactly the same exception that we put in? (Or is it not?)

Reply:
True. fixed to         assert urlerror == exc.value